fs/ftpfs.py

```diff
@@ -82,7 +83,7 @@ def ftp_errors(fs, path=None):
                 msg=message
             )
         elif code in ('501', '550'):
-            raise errors.ResourceNotFound(path=path)
+            raise errors.ResourceNotFound(path=cast(str, path))
```

Comment:
`path` may be None, but in practice this error should only occur if there is a path.

Reply:
Good catch !